Clinical-Genomics/clinical_EPPs
# TWIST WF 3

This PR adds and changes scripts that will be used in the new TWIST workflow v3. 

See the validation of the new workflow in am documented in 2145:2

**How to test**:
Install this branch on https://clinical-lims-stage.scilifelab.se
For each script:
- Run it through the lims interface in the step(s) where it's supposed to be used. 
- Verify that it behaves as expected according to the changes requested by lab.


**Scripts:**
- [x] EPPs/calculate_amount_twist.py 
 - Run in **Qubit QC (DNA) TWIST**. Button Name: **Calculate Amounts**. Test documented in 2145:2 /CS
-  Run in **Quantit QC (DNA) TWIST**. Button Name: **Calculate Amounts**. Test documented in 2145:2 /CS
- [x] EPPs/calculate_beads.py 
 - Run in **Buffer Exchange TWIST**. Button Name: **Calculate Volumes**. Test documented in 2145:2 /CS
- [x] EPPs/calculations_twist.py 
 - Run in **Aliquot samples for enzymatic fragmentation TWIST**. Button Name: **Calculate volumes**.  Test documented in 2145:2 /CS
- Run in **Qubit QC (Library Validation) TWIST**. Button Name: **Calculate volumes**.  Test documented in 2145:2 /CS
- [x] EPPs/check_kapa_amount.py 
 - Run in **Aliquot samples for enzymatic fragmentation TWIST**. Button Name: **Checking Amount needed (ng)**.  Test documented in 2145:2 /CS
- [x] EPPs/copyUDFs_from_aggregateQC.py 
 - Run in **Aliquot samples for enzymatic fragmentation TWIST**. Button Name: **1. Copy UDFs from AggregateQC - Twist**.  Test documented in 2145:2 /CS
- [x] EPPs/get_volumes_from_buffer_twist.py  
 - Run in **Qubit QC (DNA) TWIST**. Button Name: **Get Volume Elution**.  Test documented in 2145:2 /CS
- Run in **Quantit QC (DNA) TWIST**. Button Name: **Get Volume Elution**.  Test documented in 2145:2 /CS
- [x] EPPs/make_bravo_csv.py 
 - Run in **Aliquot samples for enzymatic fragmentation TWIST**. Button Name: **Generate csv for Bravo (doc 1564)**. Test documented in 2145:2 /CS
- [x] EPPs/make_kapa_txt.py
 - Run in **Enzymatic fragmentation TWIST**. Button Name: **File for Hamilton**. Test documented in 2145:2 /CS
 - Run in **KAPA Library Preparation**. Button Name: **File for Hamilton**. Test documented in 2145:2 /CS
- [x] EPPs/make_placement_map.py 
 - Run in **Buffer Exchange TWIST**. Button Name: **Make Placement map**. Test documented in 2145:2 /CS
 - Run in **Aliquot samples for enzymatic fragmentation TWIST**. Button Name: **Make placement map**.  Test documented in 2145:2 /CS
 - Run in **KAPA Library Preparation**. Button Name: **Make placement map**. Test documented in 2145:2 /CS
- [x] EPPs/make_twist_pooling_map.py
- Run in **Pool samples for hybridization TWIST**. Button Name: **Make pooling map**. Test documented in 2145:2 /CS
- [x] EPPs/pool_calculations_twist.py
- Run in **Pool samples for hybridization TWIST**. Button Name: **Calculate volumes**. Test documented in 2145:2 /CS


**Before Deploy**
For each script:
- Run on production: `python clinical_EPPs/check_configuration.py -s <script name>` to se if the script is being used in any other steps.
- Double check that the changes made to the script will not effect the calculations in other steps.
- Change configurations to the other steps where the script is being used, if needed.

- [x] EPPS/calculate_amount_twist.py - New script.
- [x] EPPs/calculate_beads.py - Only used in **Buffer Exchange v1** - No other steps effected
- [x] EPPs/calculations_twist.py - Used in many steps. See comments below
- [x] EPPs/check_kapa_amount.py - New script
- [x] EPPs/copyUDFs_from_aggregateQC.py - Used in many steps. See comments below
- [x] EPPs/get_volumes_from_buffer_twist.py - New script
- [x] EPPs/make_bravo_csv.py - Used in many steps. See comments below
- [x] EPPs/make_kapa_txt.py - New script
- [x] EPPs/make_placement_map.py  - Used in many steps. See comments below
- [x] EPPs/make_twist_pooling_map.py - Not being used in any step - No steps effected
- [x] EPPs/pool_calculations_twist.py - New script


**Review:**
- [x] tests executed by @cesve @mayabrandi 
- [x] "Merge and deploy" approved by @mayabrandi @cesve 
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
